students/ian_letourneau/Lesson07/html_render.py

```diff
@@ -0,0 +1,224 @@
+#!/usr/bin/env python3
+# Ian Letourneau
+# 5/30/2018
+
+from io import StringIO
+
+"""A script to define classes to later be initialized by the
+run_html_render.py script."""
+
+
+class Element:
+    """Parent class for all html elements to be created."""
+
+    def __init__(self, open_tag="<html>",
+                 close_tag="</html>", content=None, **kwargs):
+        """Constructor for Element objects.
+
+        :param open_tag="<html>": set opening tag parameter to html
+        should none be given 
+
+        :param close_tag="</html>": set closing tag parameter to html
+        should none be given 
+
+        :param content=None: set content value to NoneType if no 
+        content associated"""
+
+        self.open_tag = open_tag
+        self.close_tag = close_tag
+        self.content = [content]
+        # Loop through input keywords to incoporate into element
+        # attributes.
+        if kwargs:
+            self.open_tag = open_tag.strip(">")
+            for key in kwargs:
+                self.open_tag += ' {}="{}"'.format(key, kwargs[key])
+            self.open_tag += ">"
+
+    def append(self, content=None):
+        """A function to append objects or strings to higher level
+        html element objects."""
+        if hasattr(content, 'render') and self.content:
+            self.content.append(content)
+        elif hasattr(content, 'render'):
+            self.content = content
+        elif content and self.content:
+            self.content += content
+        elif content:
+            self.content = content
+
+    def render(self, file_out, cur_ind=""):
+        """A function to render page elements into stringIO object, 
+        creating html file with associated elements and formatting."""
+        self.cur_ind = "\n" + cur_ind
+        file_out.write(self.cur_ind)
+        file_out.write(self.open_tag)
+        for obj in range(len(self.content)):
+            if self.content[obj]:
+                if hasattr(self.content[obj], 'render'):
+                    self.content[obj].render(file_out, (cur_ind + "    "))
+                else:
+                    file_out.write(self.content[obj])
+        file_out.write(self.cur_ind + self.close_tag)
+
+
+class Html(Element):
+    """Html subclass of Element. Changes tags to 'html' tags."""
+
+    def __init__(self):
+        """Html object constructor."""
+        open_tag = "<html>"
+        close_tag = "</html>"
+        Element.__init__(self, open_tag, close_tag)
```

Comment:
refer to my comment above, this is not needed and could be accomplished by simply defining `tag` class variable in this class.